### ISSUE TITLE
nan loss in experiment siamrpn_resnet

### ISSUE THREAD

**Issue by yl1991** [foolwood/SiamMask#51]
Hi Qiang,
After training siamrpn for about 200 iterations I spotted `nan` in the loss. Do you think it is the problem of data preprocessing or somewhere in the code? Thank you.
Full training log attached below.
Best wishes,
Yiming

```

[2019-05-30 20:07:36,388-rk0-train_siamrpn.py#281] Epoch: [1][140/2343] lr: 0.001000    batch_time: 0.447263 (0.852683) data_time: 0.000222 (0.096190)  rpn_cls_loss: 0.348680 (0.452594)       rpn_loc_loss: 0.327530 (0.406526)       siamrpn_loss: 0.676210 (0.859119)
[2019-05-30 20:07:36,389-rk0-log_helper.py# 97] Progress: 140 / 46860 [0%], Speed: 0.853 s/iter, ETA 0:11:03 (D:H:M)

[2019-05-30 20:07:41,014-rk0-train_siamrpn.py#281] Epoch: [1][150/2343] lr: 0.001000    batch_time: 0.428898 (0.826568) data_time: 0.000247 (0.089822)  rpn_cls_loss: 0.330786 (0.445250)       rpn_loc_loss: 0.339954 (0.400595)       siamrpn_loss: 0.670740 (0.845844)
[2019-05-30 20:07:41,014-rk0-log_helper.py# 97] Progress: 150 / 46860 [0%], Speed: 0.827 s/iter, ETA 0:10:43 (D:H:M)

[2019-05-30 20:07:45,451-rk0-train_siamrpn.py#281] Epoch: [1][160/2343] lr: 0.001000    batch_time: 0.491518 (0.802561) data_time: 0.000388 (0.084235)  rpn_cls_loss: 0.332730 (0.438211)       rpn_loc_loss: 0.321817 (0.395307)       siamrpn_loss: 0.654547 (0.833518)
[2019-05-30 20:07:45,451-rk0-log_helper.py# 97] Progress: 160 / 46860 [0%], Speed: 0.803 s/iter, ETA 0:10:24 (D:H:M)

[2019-05-30 20:07:50,009-rk0-train_siamrpn.py#281] Epoch: [1][170/2343] lr: 0.001000    batch_time: 0.449069 (0.782116) data_time: 0.000453 (0.079310)  rpn_cls_loss: 0.321972 (0.431562)       rpn_loc_loss: 0.312462 (0.390696)       siamrpn_loss: 0.634434 (0.822258)
[2019-05-30 20:07:50,009-rk0-log_helper.py# 97] Progress: 170 / 46860 [0%], Speed: 0.782 s/iter, ETA 0:10:08 (D:H:M)

[2019-05-30 20:07:54,762-rk0-train_siamrpn.py#281] Epoch: [1][180/2343] lr: 0.001000    batch_time: 0.511367 (0.764989) data_time: 0.000424 (0.074935)  rpn_cls_loss: 0.293845 (0.425333)       rpn_loc_loss: 0.274334 (nan)    siamrpn_loss: 0.568179 (nan)
[2019-05-30 20:07:54,763-rk0-log_helper.py# 97] Progress: 180 / 46860 [0%], Speed: 0.765 s/iter, ETA 0:09:55 (D:H:M)

[2019-05-30 20:07:59,350-rk0-train_siamrpn.py#281] Epoch: [1][190/2343] lr: 0.001000    batch_time: 0.401666 (0.748833) data_time: 0.000397 (0.071019)  rpn_cls_loss: 0.301740 (0.419603)       rpn_loc_loss: 0.295769 (nan)    siamrpn_loss: 0.597508 (nan)
```

**Comment by XiaoCode-er** [comment]
@yl1991 I met the same problem when i trained SiamMask model only with COCO dataset.  The nan appeared in rpn_loc_loss and siammask_loss in about the 320th iteration. Then i also spotted nan in mask_iou_mean in the 18th epoch.

**Comment by yl1991** [comment]
> @yl1991 I met the same problem when i trained SiamMask model only with COCO dataset. The nan appeared in rpn_loc_loss and siammask_loss in about the 320th iteration. Then i also spotted nan in mask_iou_mean in the 18th epoch.

I think it may be caused by errors in the annotations, although the training script itself is able to check some errors, there might be more. See below. We may need to set more rules to exclude incorrect annotations in training  data. @foolwood 

![image](https://user-images.githubusercontent.com/19656705/58693818-7c427880-8389-11e9-8456-1766d4e84956.png)

**Comment by needsay** [comment]
I have found a solution.

You can modify the 117 line of SiamMask/datasets/siam_mask_dataset.py, from `if w == 0 or h == 0:` to `if w <= 0 or h <= 0:`.

This will correctly exclude incorrect annotations in COCO.

**Comment by xyl-507** [comment]
@needsay Hello, can and should the same filter be applied to DET and YT_BB? thanks！
